### PR TITLE
Allow multiple projection IDs for an event

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**

### CODE REVIEW

**Comment by alexander-yevsyukov** [comment]
@armiol this is work in progress. Tests are not added and `addIdSetFunction()`/ `removeIdSetFunction()` methods are not used because of this. Let's discuss before going further.

**Comment by alexander-yevsyukov** [comment]
@armiol 
Instead of overriding the following method...
```java
    protected Set<I> getProjectionIds(Message event, EventContext context) {
        return idSetFunctions.findAndApply(event, context);
    }
```
I would expect that a projection would add custom `IdSetFunction`s in their constructors.

File: server/src/main/java/org/spine3/server/projection/ProjectionRepository.java [hunk 1]
```diff
@@ -174,7 +219,7 @@ public void close() throws Exception {
      */
     @Override
     @Nonnull
-    @SuppressWarnings("RefusedBequest")
+    @SuppressWarnings("MethodDoesntCallSuperMethod")
```
Comment (armiol) [inline server/src/main/java/org/spine3/server/projection/ProjectionRepository.java]: Please add an explanation.

File: server/src/test/java/org/spine3/server/projection/ProjectionRepositoryShould.java [hunk 2]
```diff
@@ -249,6 +266,48 @@ private void ensureCatchesUpFromEventStorage(ProjectionRepository<ProjectId, Tes
...
+    }
+
+    @Test
+    public void obtain_id_set_function_after_put() {
+        repository.addIdSetFunction(ProjectCreated.class, idSetForCreateProject);
+
+        final Optional<IdSetFunction<ProjectId, ProjectCreated, EventContext>> out = repository.getIdSetFunction(ProjectCreated.class);
+
+        assertTrue(out.isPresent());
+        //noinspection OptionalGetWithoutIsPresent
```
Comment (armiol) [inline server/src/test/java/org/spine3/server/projection/ProjectionRepositoryShould.java]: Please use an annotation-based suppression instead. It's OK, if it is for the whole method, as this is a test codebase.

**Pull request merged**